### PR TITLE
remove all clang warnings

### PR DESCRIPTION
**Pull request by kevinkreiser**
as the title says :)

### CODE REVIEW

File: src/bindings/node/node.cc [hunk 1]
```diff
@@ -6,8 +6,6 @@
 #include <functional>
 #include <iostream>
 #include <napi.h>
-#include <rapidjson/stringbuffer.h>
-#include <rapidjson/writer.h>
```
Comment (kevinkreiser) [inline src/bindings/node/node.cc]: if you wan tto use rapidjson you should go through the utils header

File: valhalla/baldr/rapidjson_utils.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include <boost/optional.hpp>
 
 // rapidjson loves to assert and crash programs, its more useful to throw and catch
+#define RAPIDJSON_ASSERT_THROWS
```
Comment (kevinkreiser) [inline valhalla/baldr/rapidjson_utils.h]: by doing this with a newer version of rapidjson, you can remove the warning you get when their internal functions offer nothrow gaurantees but you've overridden the assertions with throws. there should be some docs on the rapidjson side about how to effectively use this. i think this is a very common use-case tbh. more info here: https://github.com/Tencent/rapidjson/issues/1280

**Comment by danpaz** [comment]
Thank you very much this cleaned up the warnings for me!

**Comment by kevinkreiser** [comment]
getting this on gcc... makes no sense to me

```
In file included from /home/kkreiser/sandbox/valhalla/valhalla/valhalla/baldr/rapidjson_utils.h:24:0,
                 from /home/kkreiser/sandbox/valhalla/valhalla/src/mjolnir/valhalla_add_predicted_traffic.cc:9:
/home/kkreiser/sandbox/valhalla/valhalla/third_party/rapidjson/include/rapidjson/pointer.h: In member function ‘bool rapidjson::GenericPointer<ValueType, Allocator>::operator<(const rapidjson::GenericPointer<ValueType, Allocator>&) const’:
/home/kkreiser/sandbox/valhalla/valhalla/third_party/rapidjson/include/rapidjson/pointer.h:407:59: error: the value of ‘i’ is not usable in a constant expression
                 return tokens_[i].length < rhs.tokens_[i].length;
                                                           ^
/home/kkreiser/sandbox/valhalla/valhalla/third_party/rapidjson/include/rapidjson/pointer.h:402:21: note: ‘std::size_t i’ is not const
         for (size_t i = 0; i < tokenCount_; i++) {
                     ^
/home/kkreiser/sandbox/valhalla/valhalla/third_party/rapidjson/include/rapidjson/pointer.h:407:35: error: parse error in template argument list
                 return tokens_[i].length < rhs.tokens_[i].length;
```

**Pull request merged**